File: 프로그래머스/lv1/12921. 소수 찾기/소수 찾기.java
import java.util.*;

class Solution {
    public int solution(int n) {
        int[] answer = new int[n + 1];
        answer[0] = 1;
        answer[1] = 1;
        
        for(int i = 2; i < Math.sqrt(n) + 1; i++){
            for(int j = 2*i; j < n + 1; j+=i){
                answer[j] = 1;
            }
        }
        return Math.toIntExact(Arrays.stream(answer).filter(a -> a != 1).count());
    }
}

File: 프로그래머스/lv1/12921. 소수 찾기/소수 찾기.js
function solution(n) {
    const answer = new Array();
    
    for(let i = 0; i < n + 1; i++){
        answer.push(i);
    }
    answer[0] = -1;
    answer[1] = -1;
    
    for(let j = 2; j < Math.sqrt(answer.length); j++){
        for(let k = 2*j; k < n + 1; k += j){
            answer[k] = -1;
        }
    }
    return answer.filter(a => a != -1).length;
}